sync-core/src/main/java/com/cloudant/http/Http.java

```diff
@@ -0,0 +1,94 @@
+package com.cloudant.http;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * Created by tomblench on 23/03/15.
+ */
+
+/**
+ * Factory methods for obtaining <code>HttpConnection</code>s.
+ *
+ * @see com.cloudant.http.HttpConnection
+ */
+public class Http {
+
+    // high level http operations, URL and URI flavoured
+
+    public static HttpConnection GET(URL url)
+    {
+        return connect("GET", null, url);
+    }
+
+    public static HttpConnection GET(URI uri)
+    {
+        return connect("GET", null, uri);
+    }
+
+    public static HttpConnection PUT(String contentType,
+                                     URL url)
+    {
+        return connect("PUT", contentType, url);
+    }
```

Comment:
I think all these methods should have the URL first. I.e.:

```java
public static HttpConnection PUT(URL url, String contentType)
```

`connect` should take it second:

```java
return connect("PUT", url, contentType);
```

As should HttpConnection constructors:

```java
return new HttpConnection(requestMethod, uri.toURL(), contentType);
```